Sets heavy GitHub CI workflows to not trigger on text documentation-only changes.

dsclear-amd opened the pull request:
## Motivation

Currently, full installation and unit test runs are conducted even for changes of only documentation files; this greatly slows down docs development work.

## Technical Details

This change modifies two of the largest workflows, azure-ci-dispatcher.yml and therock-ci.yml, as well as many rocprofiler workflows, to not trigger on PRs or pushes that consist entirely of documentation changes.

## JIRA ID

AIROCDOC-78, AIROCDOC-424, AIROCDOC-425

## Test Plan

1. Created a test branch (for example, test_branch; actual test users/dsclear/shorten_workflows_test_root) with the changes in this PR, branched from develop.
2. Branched test_branch_bin_feeder and test_branch_text_feeder from test_branch.
3. Modified test_branch_bin_feeder to add a new .h file, and submitted a PR from test_branch_bin_feeder into test_branch (https://github.com/ROCm/rocm-systems/pull/2566).
4. Modified test_branch_text_feeder to add a new .txt file, and submitted a PR from test_branch_text_feeder into test_branch (https://github.com/ROCm/rocm-systems/pull/2567).

## Test Result

The test_branch_bin_feeder PR caused the modified workflows to still trigger, as expected. The test_branch_text_feeder PR caused the modified workflows to be skipped, as desired.

dsclear-amd committed:
Sets heavy GitHub CI workflows to not trigger on docs-only changes.

Specifically, sets azure-ci-dispatcher.yml and therock-ci.yml to
	not trigger when the change consists entirely of .txt,
	.md, or .rtf files.

neon60 commented on .github/workflows/azure-ci-dispatcher.yml:
```diff
@@ -26,6 +26,10 @@ on:
       - main
       - release-staging/rocm-rel-7.*
+    paths-ignore:  # Do not trigger full tests on documentation-only changes
+      - '**.txt'
```
@dsclear-amd we need to extend and update this list.
``**/docs/**``, ``**.rst`` and others are missing.

dsclear-amd replied:
Added the additional elements.

dsclear-amd committed:
Expands list of docs-only elements excluded from GitHub CI/CD workflows.

Most notably, any changes in the 'docs' folder, or the 'docs' folder in
	any project, are now excluded.

dsclear-amd committed:
Adds some missing docs-only exclusions to workflows/therock-ci.yml.

This workflow should now correctly skip docs-only changes, even on
pushes.

dsclear-amd committed:
Merge branch 'develop' into user/dsclear/shorten_pr_checks_for_docs_changes_commitable

dsclear-amd committed:
Edits most rocprofiler workflows to exclude docs-only changes.

This should eliminate unnecessary testing, and speed up documentation
	development.

dsclear-amd committed:
Fixes some accidental changes in rocprofiler CI workflows.

neon60 commented on .github/workflows/rocprofiler-compute-continuous-integration.yml:
```diff
@@ -19,23 +19,33 @@ on:
+      - '!**.rst'
+      - '!**/.markdownlint-ci2.yaml'
+      - '!**/.readthedocs.yaml'
+      - '!**/.spellcheck.local.yaml'
```
Could we add here?

``!projects/rocprofiler-compute/docs/**``

dsclear-amd committed:
Fixes paths in altered workflows.

Specifically, fixes the old incorrect paths like '**.txt'
	to be of the style '**/*.txt'.

dsclear-amd committed:
Removes the 'modified files' folder accidentally added to previous commit.

dsclear-amd committed:
Fixes the paths-ignore in azure-ci-dispatcher.yml.

Specifically, changes them from the incorrect '**.txt' style
	to '**/*.txt'.

geomin12 reviewed (changes requested):
looks fine, although this will break PRs for `therock-ci.yml`, as TheRock CI is required to run

please add here for TheRock: https://github.com/ROCm/rocm-systems/blob/ebe22b59076479682f260cdae7246c2818c0f1d1/.github/scripts/therock_configure_ci.py#L87

dsclear-amd committed:
Moves ignore-docs-only-commits code from therock-ci workflow into its configure script.

lamb-j commented:
Is it safe to assume changes to *.txt are non-functional (and don't require testing?). For example, maybe changes to a CMakeLists.txt?

dsclear-amd commented:
> Is it safe to assume changes to *.txt are non-functional (and don't require testing?). For example, maybe changes to a CMakeLists.txt?

Excellent point, I didn't consider CMakeLists.txt. I'll remove ignoring .txt files in favour of just listing docs-specific .txt files we know to ignore.

dsclear-amd committed:
Modifies excluded paths in CI/CD workloads.

Specifically, instead of excluding all .txt files (which would catch
	things like CMakeLists.txt), now only specific .txt files
	are excluded.

dsclear-amd committed:
Restores strangely deleted section of rocprofiler-systems-continuous-integration.yml.

geomin12 reviewed (approved):
lgtm for TheRock CI

dsclear-amd committed:
Merge branch 'develop' into user/dsclear/shorten_pr_checks_for_docs_changes_commitable

dsclear-amd committed:
Changes rocprofiler workflows to use paths rather than paths-ignore.

This is because using both paths and paths-ignore at the same time
	is not allowed.
